process/transaction/shardProcess.go

```diff
@@ -503,6 +509,78 @@ func (txProc *txProcessor) processBuiltInFunctionCall(
 	return txProc.scProcessor.ExecuteBuiltInFunction(tx, acntSrc, acntDst)
 }
 
+func (txProc *txProcessor) processRelayedTxV2(
+	tx *transaction.Transaction,
+	relayerAcnt, acntDst state.UserAccountHandler,
+) (vmcommon.ReturnCode, error) {
+	if !txProc.flagRelayedTxV2.IsSet() {
+		return vmcommon.UserError, txProc.executingFailedTransaction(tx, relayerAcnt, process.ErrRelayedTxV2Disabled)
+	}
+	if tx.GetValue().Cmp(big.NewInt(0)) != 0 {
+		return vmcommon.UserError, txProc.executingFailedTransaction(tx, relayerAcnt, process.ErrRelayedTxV2ZeroVal)
+	}
+
+	_, args, err := txProc.argsParser.ParseCallData(string(tx.GetData()))
+	if err != nil {
+		return vmcommon.UserError, txProc.executingFailedTransaction(tx, relayerAcnt, err)
+	}
+	if len(args) != 4 {
+		return vmcommon.UserError, txProc.executingFailedTransaction(tx, relayerAcnt, process.ErrInvalidArguments)
+	}
+
+	userTxData := args[2]
+	userTxSig := args[3]
+
+	userTx := &transaction.Transaction{}
+	// Fill properties received as arguments
+	userTx.RcvAddr = args[0]
+	userTx.Nonce = big.NewInt(0).SetBytes(args[1]).Uint64()
+	userTx.Data = userTxData
+	userTx.Signature = userTxSig
+	userTx.Value = big.NewInt(0)
+
+	// Fill properties from main transaction
+	userTx.GasPrice = tx.GasPrice
+	totalFee, remainingFee, relayerFee, remainingGasLimit := txProc.computeRelayedTxFees(tx)
+	userTx.GasLimit = remainingGasLimit
+
+	txHash, err := core.CalculateHash(txProc.marshalizer, txProc.hasher, tx)
```

Comment:
a lot of duplicated code with the processRelayedTx function. Please extract the common functionality in a function

Reply:
done